CMake/external_libusb.cmake

```diff
@@ -20,13 +19,16 @@ ExternalProject_Add(
             -DANDROID_STL=${ANDROID_STL}
             -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libusb_install
     TEST_COMMAND ""
+    BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/libusb_install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}usb${CMAKE_STATIC_LIBRARY_SUFFIX}
 )
 
-set(LIBUSB1_LIBRARY_DIRS ${CMAKE_CURRENT_BINARY_DIR}/libusb_install/lib)
-link_directories(${LIBUSB1_LIBRARY_DIRS})
+add_library(usb INTERFACE)
+target_include_directories(usb INTERFACE ${CMAKE_CURRENT_BINARY_DIR}/third-party/libusb/libusb)
+target_link_libraries(usb INTERFACE ${CMAKE_CURRENT_BINARY_DIR}/libusb_install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}usb${CMAKE_STATIC_LIBRARY_SUFFIX})
+set(USE_EXTERNAL_USB ON) # INTERFACE libraries can't have real deps, so targets that link with usb need to also depend on libusb
 
-set(LIBUSB1_LIBRARIES usb)
-#set(LIBUSB_LOCAL_INCLUDE_PATH third-party/libusb)
-
-set(USE_EXTERNAL_USB ON)
-set(LIBUSB_LOCAL_INCLUDE_PATH ${CMAKE_CURRENT_BINARY_DIR}/third-party/libusb)
+if (APPLE)
```

Comment:
can't it be moved to `apple_config`?

Reply:
This is really part of the exported interface of `libusb`, so I think it should stay in a `libusb` specific place.  Normally this would come from a `target_*` function if we were actually compiling `libusb`, an installed cmake file, or a `pkg-config` file, but since we are using an "external project" without installation, we need to do it ourselves.

Reply:
Also, I removed (cc40ab3) `apple_config.cmake` as it wasn't being used.  :)